app/code/VConnect/Blog/Model/PostsPublishManager.php

```diff
@@ -0,0 +1,81 @@
+<?php
+declare(strict_types=1);
+
+namespace VConnect\Blog\Model;
+
+use Magento\Framework\Api\SearchCriteriaBuilder;
+use Magento\Tests\NamingConvention\true\bool;
+use VConnect\Blog\Api\Data\PostInterface;
+use VConnect\Blog\Api\PostRepositoryInterface;
+
+class PostsPublishManager
+{
+    /** @var PostInterface[] */
+    private array $noPublishedPosts = [];
+
+    private bool $postsPublishStatus;
+
+    public function __construct(
+        private PostRepositoryInterface $postRepository,
+        private SearchCriteriaBuilder $searchCriteriaBuilder,
+    ) {}
+
+    /**
+     * @throws \Magento\Framework\Exception\CouldNotSaveException
+     */
+    public function publishPosts(): void
+    {
+        if ($this->getNotPublishedPosts()) {
+            $this->setPublishTrue();
+        }
+    }
+
+    /**
+     * @return bool
+     */
+    private function getNotPublishedPosts(): bool
```

Comment:
Substitution of concepts i can see here. Your method logically says getNotPublishedPosts when i read this i expect to get exactlly posts as a result. You return true or false. I think you should return array of posts (items) . Search criteria getItems method does it. And your publishPosts method (or you can just rename it to 'execute') will expect an array of posts and proceed with them.